docs/deployment/gcloud/versions.tf

```diff
@@ -0,0 +1,15 @@
+terraform {
+  required_version = "~> 1.5"
```

Comment:
in the main you have defined one only provider `google-beta`and using it but then you bring in the second one and you do not define it in the main. either you define both and use both or pick one.